mbedtls/src/self_test.rs

```diff
@@ -58,6 +58,11 @@ pub unsafe fn disable() {
     log_f = None;
 }
 
+//
+// NOTE: The following functions if used in a multithreaded or async environment will fail.
+//       Self test functions rely on global variables to track operations and anything non-self-test related will stomp over variables.
+//       While using these, make sure no other code uses mbedtls. Multiple self test operations done asynchronously may also return failures.
+//
```

Comment:
This note only makes sense as a doc comment. Please also make change the comments to `fn enable/disable` to doc comments. Follow this template:
```
/// # Safety
///
/// text here
```